components/button/Button.tsx

```diff
@@ -0,0 +1,94 @@
+import './button.css';
+import { IButtonProps } from './button.interface';
+import { EButtonIconPosition, EButtonSizes, EButtonTypes } from './button.enum';
+import { useMemo } from 'react';
+import Icon from '../icon/Icon';
+
+/**
+ * Primary UI component for user interaction
+ */
+const Button = ({
```

Comment:
Nice example Button component. Hopefully, we can use most of it to the mumble design.